src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -630,7 +630,19 @@ void FancyZones::MoveWindowsOnDisplayChange() noexcept
 void FancyZones::UpdateDragState(require_write_lock) noexcept
 {
     const bool shift = GetAsyncKeyState(VK_SHIFT) & 0x8000;
-    m_dragEnabled = m_settings->GetSettings().shiftDrag ? shift : !shift;
+	const bool mouseR = GetAsyncKeyState(VK_RBUTTON) & 0x8000;

```

Comment:
In the ["Remarks" section of the `GetAsyncKeyState`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getasynckeystate#remarks) doc there is a note, that we also need to use `GetSystemMetrics(SM_SWAPBUTTON)` to check if the mouse buttons are not swapped.

Reply:
Great callout - didn't even know this was a thing, but it totally makes sense. So if they are swapped, VK_RBUTTON should be VK_LBUTTON, correct?

Reply:
Sounds right.